libr/anal/var.c

```diff
@@ -479,6 +505,51 @@ R_API RList *r_anal_var_list(RAnal *a, RAnalFunction *fcn, int kind) {
 					av->size = vt.size;
 					av->type = strdup (vt.type);
 					r_list_append (list, av);
+					if (dynamicVars) { // make dynamic variables like structure fields
```

Comment:
move this logic into a separate function to avoid too much nesting and make the logic more atomic and less error-prone